securetar/__init__.py

```diff
@@ -131,26 +143,66 @@ def __enter__(self) -> tarfile.TarFile:
         self._encrypt = self._aes.encryptor()
         self._padder = padding.PKCS7(BLOCK_SIZE_BITS).padder()
 
-        self._tar = tarfile.open(
-            fileobj=self,
-            mode=self._tar_mode,
-            dereference=False,
-            bufsize=self._bufsize,
-        )
-        return self._tar
-
     def __exit__(self, exc_type, exc_value, traceback) -> None:
         """Close file."""
         if self._tar:
             self._tar.close()
             self._tar = None
+        self._close_file()
+
+    def _close_file(self) -> None:
+        """Close file."""
         if self._file:
             if not self._mode.startswith("r"):
                 self._file.write(self._encrypt.update(self._padder.finalize()))
             if not self._fileobj:
                 self._file.close()
             self._file = None
 
+    @contextmanager
+    def decrypt(self, tarinfo: tarfile.TarInfo) -> Generator[BinaryIO, None, None]:
```

Comment:
We should maybe raise if this method is called when key is not set?
Maybe we should add error handling in a follow-up?